membership-attribute-service/test/services/AttributesMakerTest.scala

```diff
@@ -1,111 +1,164 @@
-        AdFree = None,
         RecurringContributionPaymentPlan = Some("Monthly Contribution"),
         MembershipJoinDate = Some(referenceDate)
       )
       )
-      AttributesMaker.attributes(testId, List(contributor, membership), referenceDate) === expected
+      AttributesMaker.zuoraAttributes(testId, List(contributor, membership), referenceDate) === expected
     }
 
     "return attributes when the membership is a friend tier" in {
-      val expected = Some(Attributes(
+      val expected = Some(ZuoraAttributes(
         UserId = testId,
         Tier = Some("Friend"),
-        MembershipNumber = None,
-        AdFree = None,
         RecurringContributionPaymentPlan = None,
         MembershipJoinDate = Some(referenceDate)
       )
       )
-      AttributesMaker.attributes(testId, List(friend), referenceDate) === expected
+      AttributesMaker.zuoraAttributes(testId, List(friend), referenceDate) === expected
+    }
+
+    "attributes" should  {
+      "return up to date Zuora attributes when they match the dynamo attributes" in {
+        val zuoraAttributes = ZuoraAttributes(
+          UserId = testId,
+          Tier = None,
+          RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+          MembershipJoinDate = None,
+          DigitalSubscriptionExpiryDate = None
+        )
+
+        val dynamoAttributes = DynamoAttributes(
+          UserId = testId,
+          Tier = None,
+          RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+          MembershipJoinDate = None,
+          DigitalSubscriptionExpiryDate = None,
+          MembershipNumber = None,
+          AdFree = None,
+          TTLTimestamp = referenceDateAsDynamoTimestamp
+        )
+
+        val expected = Some(
+          Attributes(
+            UserId = testId,
+            Tier = None,
+            RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+            MembershipJoinDate = None,
+            DigitalSubscriptionExpiryDate = None
+          )
+        )
+
+        val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), Some(dynamoAttributes))
+
+        attributes === expected
+      }
+
+      "still return Zuora attributes when the user is not in Dynamo" in {
+        val zuoraAttributes = ZuoraAttributes(
+          UserId = testId,
+          Tier = None,
+          RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+          MembershipJoinDate = None,
+          DigitalSubscriptionExpiryDate = None
+        )
+
+        val expected = Some(
+          Attributes(
+            UserId = testId,
+            Tier = None,
+            RecurringContributionPaymentPlan = Some("Monthly Contribution"),
+            MembershipJoinDate = None,
+            DigitalSubscriptionExpiryDate = None
+          )
+        )
+
+        val attributes = AttributesMaker.zuoraAttributesWithAddedDynamoFields(Some(zuoraAttributes), None)
+
```

Comment:
Could also check that the update cache function has been called exactly once in this scenario? (Or could add a separate test for this if it's not already covered elsewhere)?

Reply:
I've tried to check that here:
https://github.com/guardian/members-data-api/blob/dc8b4d6ba96baabc46c69c9e922271ca0cedcc7a/membership-attribute-service/test/services/AttributesFromZuoraTest.scala#L157

Does that cover the scenario you're thinking of?

Reply:
Yep, looks good 👍